src/check.rs

```diff
@@ -60,17 +100,128 @@ pub fn is_available(url: &Url, ctx: &CheckContext) -> Result<(), CheckError> {
         }
     }
 }
+cached_key_result! {
+    CHECK_FILE: SizedCache<String, HashSet<String>> = SizedCache::with_size(100);
+    Key = { link.to_string() };
```

Comment:
```suggestion
    /// `fetch_html` is different depending on whether the link is being loaded from disk or from the network.
    /// However, the set of URLs from disk and from the web are disjoint, so it's OK to cache based on the name of the link alone.
    Key = { link.to_string() };
```

Reply:
The `cached_key_result` macro doesn't really allow rustdoc, I've added this as regular comment.